crates/moss-cache/src/lib.rs

```diff
@@ -0,0 +1,250 @@
+use dashmap::DashMap;
+use moka::sync::{Cache, CacheBuilder};
+use moka::Expiry;
+use std::any::{type_name, Any};
+use std::fmt::Debug;
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+use thiserror::Error;
+
+type DynType = Arc<dyn Any + Send + Sync>;
+
+#[derive(Debug, Error)]
+pub enum MossCacheError {
+    #[error("The value of key '{key}' does not have type '{type_name}'")]
+    TypeMismatch { key: String, type_name: String },
+
+    #[error("The key '{key}' does not exist")]
+    NonexistentKey { key: String },
+}
+
+#[derive(Clone, Debug)]
+struct MossCacheItem {
+    pub item: DynType,
+    ttl: Duration,
+}
```

Comment:
The `Moss` prefix in type names is probably unnecessary. We’ll already be importing it from the `moss_*` crate

Reply:
Good idea! I was thinking about this too!